ddclient.in

```diff
@@ -2276,44 +2276,37 @@ sub extract_ipv4 {
     return $ip;
 }
 
+######################################################################
+## Regex to find IPv6 in standard or compressed format. Mixed IPv6/IPv4
+## not supported. Accepts leading zeros in the address. Adapted from
+## regex in Regular Expressions Cookbook by J.Goyvaerets and S.Levithan
+######################################################################
+my $regex_ipv6 = "(?<![:.0-9A-Z])              ## Negative lookbehind for word boundry
```

Comment:
Can't.  The regex itself has the boundary checks (lookaheads) embedded in the middle of the regex not just at the start and end.  In other words they are needed for the `is_ipv6` case.  If all we had to do was test for rather than extract as well then we could use a simpler boundry check, but then we would have two different regex's.  I think it is better to have just one.